### DISCUSSION TITLE
Should `SourceCodePositioner` (and other classes) consider all kinds of line breaks?

### DISCUSSION THREAD

**Discussion by Marcono1234** [pmd/pmd#3972] [Q&A] [answered]
Currently the PMD class `SourceCodePositioner` only considers `\n` as line break ([here](https://github.com/pmd/pmd/blob/9d23d79802a963e4a4edee8ec7cfcf61cc1f4155/pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/SourceCodePositioner.java#L27)). However, usually all JDK classes consider `\n`, `\r` and `\r\n` as line breaks, for example [`BufferedReader.readLine()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/BufferedReader.html#readLine()) or [`Files.readAllLines`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/Files.html#readAllLines(java.nio.file.Path,java.nio.charset.Charset)).

Would it make sense to adjust `SourceCodePositioner` (and maybe other PMD classes) to behave the same way? Or do you think that this is not worth it because it makes the code more error-prone and `\n` is the most common line break?

**Comment by oowekyala** [comment, marked as answer]
PMD 7 does consider all of `\n`, `\r`, and `\r\n` - the PR is still pending: https://github.com/pmd/pmd/pull/3893#discussion_r861740322

I'm not sure it makes sense to change this in PMD 6. SourceCodePositioner does work with `\r\n` because splitting lines on `\n` also splits correctly `\r\n` (see https://github.com/pmd/pmd/pull/3893#discussion_r861509609), and so text coordinates work the same. It doesn't support single `\r`, but that's AFAIK only used on machines that are relatively rare nowadays: https://en.wikipedia.org/wiki/Newline#Representation
